modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricerTest.java

```diff
@@ -89,8 +89,7 @@ public Payment upfrontPayment(ResolvedFixedCouponBondTrade trade) {
           return Payment.of(CurrencyAmount.zero(trade.getProduct().getCurrency()), SETTLEMENT);
         }
       };
-  private static final DiscountingFixedCouponBondProductPricer PRODUCT_PRICER =
-      DiscountingFixedCouponBondProductPricer.DEFAULT;
+  private static final DiscountingFixedCouponBondProductPricer PRODUCT_PRICER = TRADE_PRICER.getProductPricer();
   private static final DiscountingPaymentPricer PRICER_NOMINAL = DiscountingPaymentPricer.DEFAULT;
   private static final DiscountingFixedCouponBondPaymentPeriodPricer COUPON_PRICER =
       DiscountingFixedCouponBondPaymentPeriodPricer.DEFAULT;
```

Comment:
Add a test for the new "settlementDate" method.

Reply:
Added tests